### ISSUE TITLE
@RepositoryRestResource.path should allow more than one path segment [DATAREST-1520]

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-data-rest#1877]
**[alienisty](https://jira.spring.io/secure/ViewProfile.jspa?name=alienisty)** opened **[DATAREST-1520](https://jira.spring.io/browse/DATAREST-1520?redirect=false)** and commented

When creating complex applications, it is useful to group paths related to common concerns, contexts, etc.

Say we have a module "foo" and I want to group all the resources under "foo", I should be able to specify the path as "foo/resource".

I don't see the reason for the current limitation, at least for `@RepositoryRestResource`

---

**Affects:** 3.3 RC2 (Neumann)

**Comment by brodziakm** [comment]
I agree, this would be extremely useful, and don't understand why we need to limit ourselves to flat resources.

**Comment by odrotbohm** [comment]
Unfortunately that's not possible at the moment as the controllers handling the invocations use `/{repository}/…` as mapping segment and those are only mapped to single path segments.

**Comment by tayfunyasar** [comment]
I checked old issues too, I've seen that, this feature request still up to date.
